diff --git a/ShareThisForm.js b/ShareThisForm.js
index fdc734e..0a39b93 100644
--- a/ShareThisForm.js
+++ b/ShareThisForm.js
@@ -8,7 +8,8 @@
   // Adding use strict as function encapsulation ECMA 6.
   'use strict';
 
-  var drupal_st = {
+  // @see : http://stackoverflow.com/a/7048295/377342
+  window.drupal_st = {
     // These are handlerd for updating the widget pic class.
     multiW: function () {
       jQuery('.st_widgetPic').addClass('st_multi');
diff --git a/sharethis.admin.inc b/sharethis.admin.inc
index 94b8adf..8a97133 100644
--- a/sharethis.admin.inc
+++ b/sharethis.admin.inc
@@ -67,9 +67,9 @@ function sharethis_configuration_form($form, &$form_state) {
     '#required' => TRUE,
     '#type' => 'textfield',
     '#prefix' => '<div>',
-    '#suffix' => '</div><div id="myPicker"></div><script type="text/javascript">stlib_picker.setupPicker(jQuery("#myPicker"), [' . $service_string_markup . '], drupal_st.serviceCallback);</script>',
+    '#suffix' => '</div><div id="myPicker"></div><script type="text/javascript">window.stlib_picker.setupPicker(jQuery("#myPicker"), [' . $service_string_markup . '], window.drupal_st.serviceCallback);</script>', // @see : http://stackoverflow.com/a/7048295/377342
     '#title' => t("Choose Your Services."),
-    '#default_value' => t("@service_string", array('@service_string' => $service_string)),
+    '#default_value' => $service_string,
     '#maxlength' => 1024,
   );
   $form['options']['sharethis_option_extras'] = array(
diff --git a/stlib_picker.js b/stlib_picker.js
index 4b1cfdd..8b6daea 100644
--- a/stlib_picker.js
+++ b/stlib_picker.js
@@ -12,7 +12,8 @@
   // Adding use strict as function encapsulation ECMA 6.
   'use strict';
 
-  var stlib_picker = {};
+  // @see : http://stackoverflow.com/a/7048295/377342
+  window.stlib_picker = {};
   var isSecure = document.location.protocol === 'https:' ? true : false;
 
   stlib_picker.pickerList = [];
@@ -40,10 +41,10 @@
     // Append the three divs that are needed:
     jQElement.append('<div class="stp_pickerLeft"><span class="stp_header">Selected Service</span><ul class="stp_ulLeft"></ul></div>');
     jQElement.append('<div class="stp_pickerArrow">' +
-                      '<div class="stp_arrow"><img class="stp_up" src="//www.sharethis.com/images/Direction_Icons_Sprite.png"></img></div>' +
-                      '<div class="stp_arrow"><img class="stp_left" src="//www.sharethis.com/images/Direction_Icons_Sprite.png"></img></div>' +
-                      '<div class="stp_arrow"><img class="stp_right" src="//www.sharethis.com/images/Direction_Icons_Sprite.png"></img></div>' +
-                      '<div class="stp_arrow"><img class="stp_down" src="//www.sharethis.com/images/Direction_Icons_Sprite.png"></img></div>' +
+                      '<div class="stp_arrow"><img class="stp_up" src="//www.sharethis.com/images/Direction_Icons_Sprite.png" /></div>' +
+                      '<div class="stp_arrow"><img class="stp_left" src="//www.sharethis.com/images/Direction_Icons_Sprite.png" /></div>' +
+                      '<div class="stp_arrow"><img class="stp_right" src="//www.sharethis.com/images/Direction_Icons_Sprite.png" /></div>' +
+                      '<div class="stp_arrow"><img class="stp_down" src="//www.sharethis.com/images/Direction_Icons_Sprite.png" /></div>' +
                       '</div>');
     jQElement.append('<div class="stp_pickerRight"><span class="stp_header">Possible Services</span><ul class="stp_ulRight"></ul></div>');
     jQElement.append('<div class="stp_clear"></div>');
@@ -62,21 +63,21 @@
       if (jQuery.inArray(key, pickerDefaults) === -1) {
         var ul = jQElement.children('.stp_pickerRight').children('.stp_ulRight');
         if (isSecure) {
-          ul.append('<li id="st_li_"' + key + 'class="stp_li"><img src="https://ws.sharethis.com/images/"' + key + '_32.png></img><span class="stp_liText">' + value.title + '</span></li>');
+          ul.append('<li id="st_li_' + key + '" class="stp_li"><img src="https://ws.sharethis.com/images/' + key + '_32.png"></img><span class="stp_liText">' + value.title + '</span></li>');
         }
         else {
-          ul.append('<li id="st_li_"' + key + 'class="stp_li"><img src="http://w.sharethis.com/images/"' + key + '_32.png></img><span class="stp_liText">' + value.title + '</span></li>');
+          ul.append('<li id="st_li_' + key + '" class="stp_li"><img src="http://w.sharethis.com/images/' + key + '_32.png"></img><span class="stp_liText">' + value.title + '</span></li>');
         }
 
       }
     });
-    for (i = 0; i < pickerDefaults.length; i++) {
+    for (var i = 0; i < pickerDefaults.length; i++) {
       var ul = jQElement.children('.stp_pickerLeft').children('.stp_ulLeft');
       if (isSecure) {
-        ul.append('<li id="st_li_"' + pickerDefaults[i] + ' class="stp_li"><img src="https://ws.sharethis.com/images/"' + pickerDefaults[i] + '_32.png></img><span class="stp_liText">' + _all_services[pickerDefaults[i]].title + '</span></li>');
+        ul.append('<li id="st_li_' + pickerDefaults[i] + '" class="stp_li"><img src="https://ws.sharethis.com/images/' + pickerDefaults[i] + '_32.png"></img><span class="stp_liText">' + _all_services[pickerDefaults[i]].title + '</span></li>');
       }
       else {
-        ul.append('<li id="st_li_"' + pickerDefaults[i] + ' class="stp_li"><img src="http://w.sharethis.com/images/"' + pickerDefaults[i] + '_32.png></img><span class="stp_liText">' + _all_services[pickerDefaults[i]].title + '</span></li>');
+        ul.append('<li id="st_li_' + pickerDefaults[i] + '" class="stp_li"><img src="http://w.sharethis.com/images/' + pickerDefaults[i] + '_32.png"></img><span class="stp_liText">' + _all_services[pickerDefaults[i]].title + '</span></li>');
       }
     }
 
